### PR TITLE
add ability to change security group rules for vpc sg's

### PR DESCRIPTION
**Pull request by richchang0**
example output:
dalton.py -d --vpc vpc-456 dalton/templates prod us-west-2
2015-12-03 17:15:02 INFO dalton.updater (update_security_group_rules:21) Adding rules to group nat_monitor_sg in us-west-2 vpc-456: ['tcp port 8008 133.17.133.37/32']
2015-12-03 17:15:03 INFO dalton.ec2 (authorize_ingress_rules:58) Request would have succeeded, but DryRun flag is set.
2015-12-03 17:15:03 INFO dalton.updater (create_security_group_if_not_exists:29) Creating security group rich in us-west-2 vpc-456
2015-12-03 17:15:04 INFO dalton.ec2 (create:27) Request would have succeeded, but DryRun flag is set.

the file for vpc will be named like this: 
dalton/templates/prod/security_groups_us-west-2_vpc-456.yaml

### CODE REVIEW

File: dalton/ec2.py [hunk 1]
```diff
@@ -97,6 +98,6 @@ def _ec2_grant_to_dalton_rule(cls, rule, grant):
         elif rule.ip_protocol == 'icmp' and rule.from_port != '-1' and rule.to_port == '-1':
             from_port, to_port = int(rule.from_port), int(rule.from_port)
         else:
-            from_port, to_port = int(rule.from_port), int(rule.to_port)
+              from_port, to_port = int(rule.from_port), int(rule.to_port)
```
Comment (codyaray) [inline dalton/ec2.py]: I believe @chicagobuss would say "extra red!#@?#!" if this were reviewboard.

File: dalton/ec2.py [hunk 2]
```diff
@@ -20,48 +20,48 @@ def __init__(self, credentials=None, connection=None):
         self.credentials = credentials
         self.connection = connection
 
-    def create(self, group_name, description, region, vpc=None, dry_run=False):
+    def create(self, group_name, description, region, vpc_id=None, dry_run=False):
```
Comment (codyaray) [inline dalton/ec2.py]: I remember going down this path.. thought there was some issue with mixing group names (like "default" or "tagserve") with group-ids and vpc-ids. Or maybe it was that using group-names no longer works in a vpc? Anyone, lgtm if it works. :)

**Comment by codyaray** [comment]
image

File: dalton/ec2.py (hunk 2 above)
Comment (richchang0) [inline dalton/ec2.py]: hi cody!!

yeah, you can't do the `self._client(region, vpc_id).delete_security_group()` call with a group_name for vpc's. but it did work with for both ec2 and vpc's if you referenced it by group_id instead (see line 34)

Comment (codyaray) [inline dalton/ec2.py]: Beautiful! :)

File: dalton.py [hunk 3]
```diff
@@ -23,30 +23,31 @@
...
+        updater.update_security_group_rules(name, security_group.rules, region, vpc_id, prune=security_group.prune, dry_run=dry_run)
 
     # Delete any groups that aren't listed in the ruleset config
-    updater.delete_security_group_if(lambda group: group.name not in security_groups, region, vpc=None, dry_run=dry_run)
+
+    updater.delete_security_group_if(lambda group: group.name not in security_groups, region, vpc_id, dry_run=dry_run)
 
 
 if __name__ == '__main__':
     options = docopt(__doc__, version='Dalton 0.1.0')
```
Comment (bkochendorfer) [inline dalton.py]: Bump the version

File: dalton/ec2.py (hunk 1 above)
Comment (bkochendorfer) [inline dalton/ec2.py]: :+1:

File: dalton/tests/test_updater.py [hunk 4]
```diff
@@ -25,10 +24,13 @@ def test_update_security_group_rules_authorize_rules(self):
         self.expectAndReturn(service.get_all(data_center, vpc), [sg1, sg2])
         service.authorize_ingress_rules(security_group, required_group_rules, data_center, vpc, dry_run)
         self.replay()
-        updater.update_security_group_rules(security_group, required_group_rules, data_center)
+        updater.update_security_group_rules(security_group, required_group_rules, data_center, vpc)
         self.verify()
 
-    def test_update_security_group_rules_revoke_rules(self):
+    def test_update_security_group_rules_authorize_rules_in_vpc(self):
+        self.test_update_security_group_rules_authorize_rules(vpc="brettifer")
```
Comment (bkochendorfer) [inline dalton/tests/test_updater.py]: :blush: I like this pattern

File: dalton/tests/test_updater.py [hunk 5]
```diff
@@ -45,7 +47,11 @@ def test_update_security_group_rules_revoke_rules(self):
         updater.update_security_group_rules(security_group, frozenset(), data_center, dry_run=dry_run)
         self.verify()
 
-    def test_update_security_group_rules_authorize_and_revoke_rules(self):
+    def test_update_security_group_rules_revoke_rules_in_vpc(self):
+        self.test_update_security_group_rules_revoke_rules(vpc="konkolnafulous")
+
```
Comment (bkochendorfer) [inline dalton/tests/test_updater.py]: Extra space

File: dalton/tests/test_updater.py [hunk 6]
```diff
@@ -45,7 +47,11 @@ def test_update_security_group_rules_revoke_rules(self):
         updater.update_security_group_rules(security_group, frozenset(), data_center, dry_run=dry_run)
         self.verify()
 
-    def test_update_security_group_rules_authorize_and_revoke_rules(self):
+    def test_update_security_group_rules_revoke_rules_in_vpc(self):
+        self.test_update_security_group_rules_revoke_rules(vpc="konkolnafulous")
```
Comment (bkochendorfer) [inline dalton/tests/test_updater.py]: @akonkol

**Comment by bkochendorfer** [comment]
:shipit:

**Comment by akonkol** [comment]
lgtm, shippity ship ship

**Comment by richchang0** [comment]
@codyaray you might be the only one with write access to this repo. can you add some of us as collaborators haha

**Comment by codyaray** [comment]
I actually don't think I have write/admin access to this repo anymore. Its cool though... I mean, that's what open source is all about, right? ;)

**Pull request merged**